File: src/main/java/com/myapp/controller/FeedController.java
package com.myapp.controller;

import com.myapp.dto.PageParams;
import com.myapp.dto.PostDTO;
import com.myapp.service.MicropostService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/feed")
public class FeedController {

    @SuppressWarnings("UnusedDeclaration")
    private static final Logger logger = LoggerFactory.getLogger(FeedController.class);

    private final MicropostService micropostService;

    @Autowired
    public FeedController(MicropostService micropostService) {
        this.micropostService = micropostService;
    }

    @RequestMapping(method = RequestMethod.GET)
    public List<PostDTO> feed(PageParams pageParams) {
        return micropostService.findAsFeed(pageParams);
    }

}

File: src/main/kotlin/com/myapp/controller/FeedController.kt
package com.myapp.controller

import com.myapp.domain.Micropost
import com.myapp.dto.request.PageParams
import com.myapp.service.FeedService
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/feed")
class FeedController(
    private val feedService: FeedService
) {

    @Suppress("unused")
    private val logger = LoggerFactory.getLogger(FeedController::class.java)

    @GetMapping
    fun feed(
        @RequestParam(required = false) sinceId: Long?,
        @RequestParam(required = false) maxId: Long?,
        @RequestParam(required = false, defaultValue = "20") count: Int
    ): List<Micropost> {
        return feedService.findFeed(PageParams(sinceId, maxId, count))
    }

}